app/namespaces/data_provider/data_provider_controller.py

```diff
@@ -79,8 +81,8 @@ def post(self):
             result = jitter_pins(result)
 
         # Only paginate if all the pagination parameters have been specified
-        if page_index is not None and per_page is not None and sorting_key is not None and reverse is not None:
-            result = get_paginated_records(result, sorting_key, page_index, per_page, reverse)
+        if min_page_index is not None and max_page_index is not None and per_page is not None and sorting_key is not None and reverse is not None:
```

Comment:
Do we need sorting_key and reverse to be specified in order to have to paginate. Maybe we can have default values (e.g. reverse should default to false, sorting_key could maybe default to sampling_end_date???)